src/sentry/workflow_engine/handlers/condition/event_frequency_base_handler.py

```diff
@@ -30,16 +30,14 @@ class _QSTypedDict(TypedDict):
 
 
 class BaseEventFrequencyConditionHandler(ABC):
-    @property
+    intervals: ClassVar[dict[str, tuple[str, timedelta]]] = STANDARD_INTERVALS
+
+    @classmethod
     @abstractmethod
-    def base_handler(self) -> Self:
+    def get_base_handler(cls) -> type["BaseEventFrequencyConditionHandler"]:
```

Comment:
having some trouble not having this be a `classmethod`. if i make it a `ClassVar`, i get `TypeError: Can't instantiate abstract class BaseEventFrequencyConditionHandler without an implementation for abstract method 'batch_query'` when calling `handler().get_rate_bulk`

Reply:
i updated the methods to be `classmethods` and now the issue is, i cannot make a ClassVar = class inside of the same class because the class has not been instantiated yet, so it seems like the function is the way to go